classes/client/object_client.php

```diff
@@ -42,19 +42,57 @@ public function register_stream_wrapper() {
      *
      * @return bool.
      */
-    public function support_signed_urls() {
+    public function support_presigned_urls() {
         return false;
     }
 
     /**
      * Generates pre-signed URL to storage file from its hash.
      *
      * @param string $contenthash File content hash.
+     * @param array $headers request headers.
      *
      * @throws \coding_exception
      */
-    public function generate_signed_url($contenthash, $filename) {
+    public function generate_presigned_url($contenthash, $headers) {
         throw new \coding_exception("Pre-signed URLs not supported");
     }
 
+    /**
+     * Checks that Content-Disposition represented in headers.
+     *
+     * @param array $headers request headers.
+     *
+     * @return bool
+     */
+    public function content_disposition_exists_in_headers($headers) {
+        foreach ($headers as $header) {
+            $found = strpos($header, 'Content-Disposition');
```

Comment:
It's more complicated than this, it's not a boolean header.

```
Content-Disposition: attachment; filename="cool.html"
Content-Disposition: inline

```

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition

We need to actually parse the contents of the header, pull it apart, and then reuse parts of it in difference ways:

1) the filename="filename.jpg" page, we could use this instead of parsing the request url. Really we should do both, and this one takes precendence

2) the 'attachment' vs 'inline', we should pass this along to aws to. Ideally if we can pass the entire header over the AWS and get them to add it? no idea if this is possible